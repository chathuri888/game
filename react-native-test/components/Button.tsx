import React from "react";
import { Pressable } from "react-native";
import AppText from "./AppText";

function Button({ styles, children, onPress, testStyle }: Props) {
    return (
        <Pressable onPress={() => onPress()} style={styles}>
            <AppText style={testStyle}>{children}</AppText>
        </Pressable>
    );
}

interface Props {
    onPress: Function,
    children: React.ReactNode,
    styles?: any,
    testStyle?: any,
    [otherProps: string]: any,
}
export default Button;