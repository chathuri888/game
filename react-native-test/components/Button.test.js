import * as React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";


describe('Button', () => {
    it('renders correctly', () => {
        const gotoNext = jest.fn();
        const tree = renderer.create(<Button text="OK" onPress={gotoNext} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
