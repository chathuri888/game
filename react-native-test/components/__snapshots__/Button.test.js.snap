// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      Array [
        Object {
          "color": "#545454",
          "fontFamily": "Avenir",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  />
</View>
`;
